packaging_automation/common_tool_methods.py

```diff
@@ -290,3 +321,114 @@ def remove_cloned_code(exec_path: str):
         except:
             print(f"Some files could not be deleted in directory {exec_path}. "
                   f"Please delete them manually or they will be deleted before next execution")
+
+
+def process_docker_template_file(project_version: str, templates_path: str, template_file_path: str):
+    minor_version = get_minor_project_version(project_version)
+    env = get_template_environment(templates_path)
+    template = env.get_template(template_file_path)
+    return f"{template.render(project_version=project_version, project_minor_version=minor_version)}\n"
+
+
+def write_to_file(content: str, dest_file_name: str):
+    with open(dest_file_name, "w") as writer:
+        writer.write(content)
+
+
+def get_gpg_fingerprint_from_name(name: str):
+    result = subprocess.run(f"gpg --list-keys ", check=True, shell=True, stdout=subprocess.PIPE)
+    lines = result.stdout.decode("ascii").splitlines()
+    counter = 0
+    line_found = False
+    for line in lines:
+        if line.startswith("uid") and name in line:
+            line_found = True
+            break
+        counter = counter + 1
+    if not line_found:
+        raise ValueError(f"Key with the name {name} could not be found ")
+    else:
+        return lines[counter - 1].strip()
+
+
+def delete_gpg_key_by_name(name: str):
+    try:
+        while True:
+            key_id = get_gpg_fingerprint_from_name(name)
+            run(f"gpg --batch --yes --delete-secret-key {key_id}")
+            run(f"gpg --batch --yes --delete-key {key_id}")
+    except ValueError:
+        print(f"Key for the name {name} does not exist")
+
+
+def get_secret_key_by_fingerprint(fingerprint: str) -> str:
+    try:
+        cmd = f'gpg --batch --export-secret-keys -a "{fingerprint}" | base64'
+        ps = subprocess.run(cmd, shell=True, stdout=subprocess.PIPE, stderr=subprocess.STDOUT, timeout=2)
+        secret_key = ps.stdout.decode("ascii")
+        return secret_key
+    except subprocess.TimeoutExpired:
+        raise ValueError(
+            f"Error while getting key. Most probably packaging key is stored with password. "
+            f"Please remove the password when storing key with fingerprint {fingerprint}")
+
+
+def get_secret_key_by_fingerprint_with_password(fingerprint: str, passphrase: str) -> str:
+    try:
+        gpg = gnupg.GPG()
+        private_key = gpg.export_keys(fingerprint, True, passphrase=passphrase)
```

Comment:
1) & 2) I think comments for `get_secret_key_by_fingerprint` apply to this function too.

Also, minor: Let's specify what are we setting to `True` (like X=True) since readers might not know about gpg's API